### ISSUE TITLE
Rule Change: no-unused-vars should detect jsdoc usage

### ISSUE THREAD

**Issue by Eccenux** [eslint/eslint#20498]
### What rule do you want to change?

no-unused-vars

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const ExtJsHelper = require("./ExtJsHelper");
const LmsHelper = require("./LmsHelper");
// eslint-disable-next-line no-unused-vars
const { Page, Browser } = require("puppeteer");

class TestHelpers {
	/**
	 * @param {Page} page pptr page.
	 * @param {Browser} browser pptr browser.
	 */
	constructor(page, browser) {
 		this.extJsHelper = new ExtJsHelper(page);
		this.lmsHelper = new LmsHelper(page, browser);
	}
}

module.exports = TestHelpers;
```

### What does the rule currently do for this code?

Shows a warning saying `Page` is not used, so I have to add `// eslint-disable-next-line no-unused-vars`.

### What will the rule do after it's changed?

Adding `// eslint-disable-next-line no-unused-vars` should not be necessary, as both variables are used in this JSDoc:
```
	/**
	 * @param {Page} page pptr page.
	 * @param {Browser} browser pptr browser.
	 */
```

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

There are cases where a variable is used in both JSDoc and regular code. In such situations, I have to remove `eslint-disable-next-line`, which is a bit inconvenient. On the other hand, the IDE or code editor won't know the variable's type unless I define and import it.

**Comment by DMartens** [comment]
I do not think we want this as part of core as:
1. We removed the deprecated JSDoc-specific rules in version 9
2. We have specific API for rules to mark variables as used

A rule to mark types as used in JSDoc is already available in [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/no-undefined-types.md#markvariablesasused).

**Comment by mdjermanovic** [comment]
I agree with @DMartens. Parsing and analyzing JSDoc is out of scope for core rules. You can enable eslint-plugin-jsdoc's rule `no-undefined-types` to address this problem.

**Comment by Eccenux** [comment]
Seems like the plugin you recommended doesn't work with the current ESLint version (which I'm using).

This issue for jsdoc: https://github.com/gajus/eslint-plugin-jsdoc/issues/1653
Is tied to typescript-eslint: https://github.com/typescript-eslint/typescript-eslint/issues/11952

So, as I understand it, this will not work for now.
